### ISSUE TITLE
How to know if "muted inline autoplay" was used?

### ISSUE THREAD

**Issue by bramus** [jwplayer/jwplayer#1597]
### The problem

Version 7.8.1 [“added support for inline autoplay while muted on iOS and Android”](https://developer.jwplayer.com/jw-player/docs/developer-guide/release_notes/release_notes_7/#781-113016).

There is however no way to programmatically know if this kind of behaviour is active or not _(or it is not documented – see #1596)_

Trying to detect this behaviour ourselves using the `play` event, some UA-sniffing, and checking of the audio levels also fails because jwplayer doesn't indicate that the video is muted:

- `jwplayer().getMute()` returns `false`
- `jwplayer().getVolume()` returns `100`

### Wanted Behavior

1. `jwplayer().getMute()` should return `true` if muted inline autoplay is used.
2. There should be no need to sniff this behaviour out ourselves. Some method or event (or some combination) to get to know if a video is playing inline while muted would be handy. That way a developer could – for example – place a muted icon somewhere in the UI and (upon click) unmute the audio.

### Steps to reproduce

1. Watch a video play inline, while muted, on an iPad running iOS10.
2. …

### Environment

- iPad/iPhone
- iOS 10
- Safari 10
- 7.8.2

**Comment by artur-michalak** [comment]
We have another issue connected to above case. 
When playback has started we'd like to set volume with setVolume method (in fact it's another way of unmute sound). 
Unfortunately it does not work because of this if https://github.com/jwplayer/jwplayer/blob/5221cb37be74c168d123315f38598975f45936c3/src/js/controller/model.js#L328. this.get('mute') returns false while it's muted on iOS

**Comment by bramus** [comment]
@artur-michalak What you describe is actually expected (and wanted) behaviour (*). One shouldn't (programmatically) unmute autoplaying videos as that would be annoying. Only after a user interaction auto may (programmatically) be unmuted.

Relevant to this issue here is that `jwplayer().getMute()` _(wrongly)_ returns `false` in case the video is mute+autoplaying. It shout return `true` in that case, so that one can show an unmute button in the UI (and then unmute upon tapping/clicking said button).

--

(*) It's also spec'd as such. See https://webkit.org/blog/6784/new-video-policies-for-ios/ (for example) for more details:

> - `<video muted>` elements will also be allowed to autoplay without a user gesture.
> - `<video>` elements will be allowed to `play()` without a user gesture if their source media contains no audio tracks, or if their `muted` property is set to `true`.
> - If a `<video>` element gains an audio track or becomes un-muted without a user gesture, playback will pause.

**Comment by bramus** [comment]
In addition to the report above: In the releasenotes I found:

> JW 7.8 also adds support for autoplay on mobile, including a new interface element to unmute autostarted videos on mobile, […]

That interface element is not visible to me (`controls` is set to `false` in my config).

**Comment by artur-michalak** [comment]
@bramus following specs wich you quoted
_If a <video> element gains an audio track or becomes un-muted without a user gesture, playback will pause._

 I did't say that i'm not doing it without user interaction. Actually i'm creating custom button for this, so i think my case is still valid.

**Comment by bramus** [comment]
@artur-michalak Aha, OK. That wasn't clear to me. Great to hear that you're (planning on) doing it the proper way too :-)

**Comment by egreaves** [comment]
@bramus, thanks for pointing this out. 

Internally, there's an `autostartMuted` property that's used to determine if the player is muted for autoplay. 

`getMute()` should return `model.get('mute') || model.get('autostartMuted')`.

**Comment by egreaves** [comment]
@artur-michalak, the use case you described should work provided the call to `setVolume()` is the result of a user interaction. We'll address this along with the issue @bramus pointed out.

**Comment by fdambrosio** [comment]
Hi, we need to disable the inline muted autoplay on mobile streams. 
I don't know if this is an issue but if I set 
```
	mute: false, 
	autostart: true
```

on Android/iOS I have an unexpected muted player. 

Demo: http://web-0.radioradicale.it/prova/player7/7.8.3-autoplay.html

**Comment by egreaves** [comment]
@fdambrosio since mobile devices require the video element to be muted in order to autoplay, we made the design decision to mute the player when `autostart: true`. `mute: false` is currently independent of autostart behavior, but we've since realized that some publishers want the ability to autostart on desktop but not on mobile because of the mute requirement. Also, to @bramus point above, the player should return the correct mute state when `getMute()` is called. We'll be addressing this soon.

In the meantime, you could do one of 2 things:

Your setup:
```
var playersito = jwplayer('playersito').setup({
	playlist: [...],
	width:544, 
	height: 306, 
	mute: false, 
	autostart: true,
	}).on('error', function(event){
			console.log('error', event);
	}).on('ready', function(){
		
	});
```

Option 1: Don't autostart on mobile if `mute === false`:

```
var playersito = jwplayer('playersito');
var autostart = !playersito.utils.isMobile();
playersito.setup({
	playlist: [...],
	width:544, 
	height: 306, 
	mute: false, 
	autostart: autostart,
	}).on('error', function(event){
			console.log('error', event);
	}).on('ready', function(){
		
	});
```

Option 2: If you want to autostart on mobile, set mute to true to keep your mute state in sync until we release a fix:

```
var playersito = jwplayer('playersito');
var mute = playersito.utils.isMobile();
playersito.setup({
	playlist: [...],
	width:544, 
	height: 306, 
	mute: mute, 
	autostart: autostart,
	}).on('error', function(event){
			console.log('error', event);
	}).on('ready', function(){
		
	});
```

**Comment by egreaves** [comment]
@bramus I forgot to mention that `video.volume` is always going to return `0` or `1` on a mobile device, so `getVolume()` will return `0` or `100`.

**Comment by bramus** [comment]
Thanks to all involved for resolving this issue this fast. 👍
